### PR TITLE
fix(bans): minor corrections

### LINKED ISSUE #12690: Error on banning call still removes participant
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

## Steps to reproduce
1. Remove a user, use the ban option
2. Leave the note empty (will be fixed on API level, but helpful to create the error in this case)
3. Remove user

### Expected behaviour
When banning fails you can try again

### Actual behaviour
Participant is still removed, since there is no option to ban participants that are not participants atm, there is no way to redo the ban.

### PR DESCRIPTION
**Pull request by Antreesy**
### ☑️ Resolves

- Fix #12690
- Ref #12735
- Ref #12293
	- Adjust text in 403 view
	- Sanitize internal note
	- Use banned and moderator display names
	- Don't remove user if ban failed
	- replace TextField with TextArea for internal note, handle max length exceed
	- handle loading state for NcDialog
	- allow to set permissions and ban federated users

### 🖼️ Screenshots / Screencasts

🏡 After
image
image
image

### 🏁 Checklist

- [x] 🌏 Tested with Chrome, Firefox and Safari or should not be risky to browser differences
- [x] 🖥️ Tested with Desktop client or should not be risky for it 
- [ ] ⛑️ Tests are included or not possible

### CODE REVIEW

File: src/components/ConversationSettings/BanSettings/BannedItem.vue [hunk 1]
```diff
@@ -54,12 +54,12 @@ export default {
 	computed: {
 		banInfo() {
 			return [
-				t('spreed', '<strong>Banned by:</strong> {actor}', { actor: this.ban.actorId },
+				t('spreed', '<strong>Banned by:</strong> {actor}', { actor: this.ban.moderatorDisplayName },
 					undefined, { escape: false, sanitize: false }),
```
Comment (nickvergessen) [inline src/components/ConversationSettings/BanSettings/BannedItem.vue:58]: Still XSS via user display name.
Please just put the html + label into the source of the frontend and add the input manually

**Comment by DorraJaouad** [comment]
Also do not close dialog when there is an error in banning or removing ? and to disable submit button on loading in order not to send multiple requests.

File: src/components/RightSidebar/Participants/Participant.vue [hunk 2]
```diff
@@ -796,10 +801,21 @@ export default {
...
 					|| this.participant.actorType === ATTENDEE.ACTOR_TYPE.EMAILS)
 		},
 
+		maxLengthWarning() {
+			if (this.internalNote.length <= 4000) {
+				return ''
+			}
+			return t('spreed', 'The text must be less than or equal to {maxLength} characters long. Your current text is {charactersCount} characters long.', {
+				maxLength: 4000,
+				charactersCount: this.internalNote.length,
```
[the remark this answers is not in the document]
Comment (Antreesy) [reply src/components/RightSidebar/Participants/Participant.vue:815]: We already have that translation ready. Don't need to make the same for different number

[the remark this answers is not in the document]
Comment (DorraJaouad) [reply src/components/RightSidebar/Participants/Participant.vue:815]: Fair point

File: src/components/RightSidebar/Participants/Participant.vue [hunk 3]
```diff
@@ -318,17 +318,21 @@
...
 				<template v-if="isBanParticipant">
-					<NcTextField v-if="isBanParticipant"
+					<NcTextArea v-if="isBanParticipant"
 						class="participant-dialog__input"
 						:label="t('spreed', 'Internal note (reason to ban)')"
+						:error="!!maxLengthWarning"
 						:value.sync="internalNote" />
+					<span v-if="maxLengthWarning" class="participant-dialog__error-hint">
+						{{ maxLengthWarning }}
+					</span>
```
[the remark this answers is not in the document]
Comment (Antreesy) [reply src/components/RightSidebar/Participants/Participant.vue]: image

File: src/components/RightSidebar/Participants/Participant.vue [hunk 4]
```diff
@@ -318,17 +318,19 @@
 					{{ t('spreed', 'Also ban from this conversation') }}
 				</NcCheckboxRadioSwitch>
 				<template v-if="isBanParticipant">
-					<NcTextField v-if="isBanParticipant"
+					<NcTextArea v-if="isBanParticipant"
 						class="participant-dialog__input"
 						:label="t('spreed', 'Internal note (reason to ban)')"
+						:error="!!maxLengthWarning"
+						:helper-text="maxLengthWarning"
 						:value.sync="internalNote" />
```
Comment (DorraJaouad) [inline src/components/RightSidebar/Participants/Participant.vue:326]: This to be fixed in upstream (specifying max width and max height) ?
image

**Review by DorraJaouad — APPROVED** [review]
it works :p

File: src/components/RightSidebar/Participants/Participant.vue (hunk 4 above)
Comment (Antreesy) [reply src/components/RightSidebar/Participants/Participant.vue:326]: `resize="vertical"` is enough to prevent overflow

**Pull request merged**